### ISSUE TITLE
Wrong exported function names & form-data generation

### ISSUE THREAD

**Issue by nichujie** [anymaniax/orval#155]
## What are the steps to reproduce this issue?
1. Use the `openapi.json` here: https://pastebin.ubuntu.com/p/XJpBv4xPH6/
2. Use `orval` to generate client files with just: `orval -i openapi.json -o test.ts`
3. Check the generated `test.ts`

### Issue 1
The function names exported seems to be different with the generated function names.
For example, if a generate method has name `redirectToDocsGet`, then at the end of file `redirect_to_docs__get` will be returned by the function, instead of its original name.

```typescript
export const getJOJHorse = () => {
  const redirectToDocsGet = <Data = unknown>(
       options?: AxiosRequestConfig
   ) => {
      return axios.get<Data extends unknown ? unknown : Data>(
        ``,options
      );
  }
  // ... some other methods 
  return {redirect_to_docs__get, ...};
};
```

### Issue 2
There's a special API in our project that uses `multipart/form-data`. And the generated function turns out to be like:

```typescript
const submitSolutionToProblemApiV1ProblemsProblemPost = <Data = unknown>(
    problem: string,
    codeFile: CodeFile, options?: AxiosRequestConfig
 ) => {
    const formData = new FormData();
    formData.append('code_type', submitSolutionToProblemApiV1ProblemsProblemPostBody.code_type)
    formData.append('file', submitSolutionToProblemApiV1ProblemsProblemPostBody.file)
    // should be codeFile.code_type & codeFile.file

    return axios.post<Data extends unknown ? RecordResp : Data>(
      `/api/v1/problems/${problem}`,
      formData,options
    );
  }
```

Instead of using the input argument `codeFile`, it uses `submitSolutionToProblemApiV1ProblemsProblemPostBody`, which you cannot find in the entire file.

## What were you expecting to happen?
1. The exported functions names should be identical as the generated ones.
2. The `form-data` requests can be generated correctly.

## Any logs, error output, etc?
I also pasted a full output of mine [here](https://pastebin.ubuntu.com/p/kDfbFTDvXs/).

## Any other comments?
1. The `openapi.json` we used is generated by `FastAPI`.
2. I strongly recommend you generate using the `openapi.json` and open the file in an IDE, which will expose the issues immediately.
3. The indentation fo the generated file is also a little bit weird.

## What versions are you using?
**Operating System:** macOS Catalina 10.15.7
**Package Version:** 5.4.3
**Browser Version:** Chrome 88.0.4324.150

**Comment by nichujie** [comment]
For the first issue, I can actually override the `operationName` method to make the generated and exported function names identical. However, it seems like the exported names will not be affected by the generated names.

I think it would make more sense to keep a list of generated function names when generating, and export them together at the end. This should be easier, and can handle whatever names generated by the users.

**Comment by anymaniax** [comment]
Hello @nichuji, your problem should be fixed with the `5.4.4` and thanks for the issue. I automatically camelCase the operation id for the name of the function that was a choice but you can use the `operationName` property to override it. If you want to handle yourself the formData you can use a [mutator](https://orval.dev/reference/configuration/output#mutator). For the formatting, there is an issue to automatically run prettier [here](#136) but at the moment you can do it yourself by running orval and then prettier just after.

**Comment by nichujie** [comment]
Thank you so much!
I've also been looking for tools that can generate client files from openapi schema. By far, orval may be the only one I've seen to support axios and react-query. Looking forward to more improvement!
